examples/aws/storage-efs/README.md

```diff
@@ -0,0 +1,242 @@
+    --region $CLUSTER_REGION \
+    --performance-mode generalPurpose \
+    --query 'FileSystemId' \
+    --output text)
+```
+
+## 5.0 Create Mount Targets for your cluster
+1. [Optional] If you are re-using an existing EFS Volume, you will first have to delete any old mount targets. You can use the following commands for this - 
+```
+aws efs describe-mount-targets --file-system-id $file_system_id
+aws efs delete-mount-target --mount-target-id <each-id>
+```
+
+2. Determine the IP address of your cluster nodes.
+```
+kubectl get nodes
+```
+
+3. Determine the IDs of the subnets in your VPC and which Availability Zone the subnet is in.
+```
+aws ec2 describe-subnets \
+    --filters "Name=vpc-id,Values=$vpc_id" \
+    --query 'Subnets[*].{SubnetId: SubnetId,AvailabilityZone: AvailabilityZone,CidrBlock: CidrBlock}' \
+    --output table
+```
+
+4. Add mount targets for the subnets that your nodes are in. If are more than 1 nodes in the cluster, you'd run the command once for a subnet in each AZ that you had a node in, replacing subnet-EXAMPLEe2ba886490 with the appropriate subnet ID from the previous command
+```
+aws efs create-mount-target \
+    --file-system-id $file_system_id \
+    --security-groups $security_group_id \
+    --subnet-id subnet-EXAMPLEe2ba886490
+```
+
+
+## 6.0 Using the EFS Storage in Kubeflow
+
+### Sample 1 - Static Provisioning
+[Using this sample from official AWS Docs](https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/master/examples/kubernetes/multiple_pods) we have provided the required spec files in the sample subdirectory but you can create the PVC another way. 
+
+1. Use the `$file_system_id` you recorded before or use the following command to get the efs filesystem id - 
+```
+aws efs describe-file-systems --query "FileSystems[*].FileSystemId" --output text
+```
+
+2. Now edit the sample/pv.yaml and edit the `volumeHandle` to point to your EFS filesystem
+
+3. Now create the required persistentvolume, persistentvolumeclaim and storageclass resources as -
+```
+kubectl apply -f examples/aws/storage-efs/sample/pv.yaml
+kubectl apply -f examples/aws/storage-efs/sample/pvc.yaml
+kubectl apply -f examples/aws/storage-efs/sample/sc.yaml
+```
+
+
+## 7.0 Connecting to Central dashboard
+Port Forward as needed and Login to http://localhost:8080 using default credentials.
+
+## 8.0 Test your Setup
+Check the `Volumes` tab in Kubeflow and you should be able to see your PVC is available for use within Kubeflow as follows - 
+
+### 8.1 Additional Permissions 
+You might need to specify some additional directory permissions on your worker node before you can use these as mount points. The set-permission-job.yaml is an example of how you could set these permissions to be able to use the efs as your workspace in your kubeflow notebook. Edit the spec file as needed for your usecase
+```
+kubectl apply -f examples/aws/storage-efs/sample/set-permission-job.yaml
+```
+
+### 8.2 Using Kubeflow Notebooks
```

Comment:
Actually EFS works as both workspace and data volume. I have specifically mentioned workspace in the text below since that is what I have used for the TFjob example, so not essential here. 

Quirk however is that FSx seems to be working only for data volume or needs some additional permissions.